docs/proposals-accepted/202206-active-series-limiting-hashring.md

```diff
@@ -0,0 +1,229 @@
+
+This approach would add validation logic within Receive Router or RouterIngestor modes and can be optionally enabled via flags.
+
+With such approach, we do not need to calculate an increase based on requests, as this will be handled by Receive instrumentation and meta-monitoring solution. We only need to query the latest HEAD series value for a tenant summed across all receives and limit remote write requests if the result of the instant query is greater than the configured limit.
+
+The value of current active series for each tenant can be cached in a map which would be updated by meta-monitoring query which is executed periodically. This map will be referred to for `latestCurrentSeries`.
+
+So if a user configures a *per-tenant* limit, say `globalSeriesLimit`, the resultant limiting equation here would simply be `globalSeriesLimit >= latestCurrentSeries` which is checked on request.
+
+<img src="../img/meta-monitoring-validator.png" alt="Meta-monitoring-based Validator" width="800"/>
+
+#### Pros:
+
+* Simpler as compared to other solutions and easier to implement
+* Lesser endpoint calls, so improved latency
+* Relies on "external to Thanos" system, and doesn’t increase load on Receive
+* Does not add much tenancy-based complexity to Thanos
+* No need to merge statistics across replicas, handled by meta-monitoring
+* Additional request-based rate limiting can be done within same component
+* In case, external meta-monitoring solution is down, can fall back to per-replica-tenant limits
+* Growing our instrumentation to improve validator, improves our observability as well
+
+#### Cons:
+
+* Not very accurate
+  * We do not know exact state of each TSDB, only know view of meta-monitoring solution, which gets updated on every scrape
+  * We do not account for how much a remote write request will increase the number of active series, only infer that from query result after the fact
+* Data replication (quorum-based) will likely cause inaccuracies in HEAD stat metrics
+* Dependence on external system, so best effort availability
+
+#### Why this is preferred?
+
+This is the simplest solution that can be implemented within Thanos Receive that can help us achieve best-effort limiting and stability. The fact that it does not rely on inter-Receive communication, which is very complex to implement, makes it a pragmatic solution.
+
+A full-fledged reference implementation of this can be found here: https://github.com/thanos-io/thanos/pull/5520.
+
+## Alternatives
+
+There are a few alternatives to what is proposed above,
+
+### Receive Router Validation
```

Comment:
Perhaps we could try to put number prefixes for easier reference when discussing this proposal

Reply:
Have added section numbers 🙂